### PR TITLE
to bring CAIP-25 draft text up to date

### LINKED ISSUE #142: CAIP-25: Is "handshake" the appropriate word for the CAIP-25 procedure?
Ref: #138 

## Problem

Participants raised the point that "handshake" may be an inappropriate term for the CAIP-25 procedure and the associated method. Specifically, "handshaking" is used in telecommunications between two parties to determine that they are using the same communications protocol. In our case, by the time a `caip_handshake` request is made, this has of course already been established.

### PR DESCRIPTION
**Pull request by bumblefudge**
If merged, this would supersede #108 (see for older discussions) and also closes #142

### CODE REVIEW

File: CAIPs/caip-25.md [hunk 1]
```diff
@@ -1,59 +1,82 @@
...
 {
     "id": 1,
     "jsonrpc": "2.0",
-    "method": "caip_handshake",
-    "params": {
-        "chains": ["eip155:1"],
-        "methods": ["eth_sendTransaction", "eth_signTransaction", "eth_sign", "personal_sign"]
-    }
+    "method": "provider_authorization",
+    "eip155": [
```
Comment (FrederikBolding) [inline CAIPs/caip-25.md:52]: These should at least be nested in `params`, maybe even in `params.namespaces`

File: CAIPs/caip-25.md [hunk 2]
```diff
@@ -1,59 +1,85 @@
...
 {
     "id": 1,
     "jsonrpc": "2.0",
-    "method": "caip_handshake",
-    "params": {
-        "chains": ["eip155:1"],
-        "methods": ["eth_sendTransaction", "eth_signTransaction", "eth_sign", "personal_sign"]
-    }
+    "method": "provider_authorization",
+    "params": [
```
Comment (FrederikBolding) [inline CAIPs/caip-25.md:52]:
```suggestion
    "params": {
```

File: CAIPs/caip-25.md [hunk 3]
```diff
@@ -1,59 +1,85 @@
...
+        "eip155": {
+            "chains": ["eip155:1"],
+            "methods": ["eth_sendTransaction", "eth_signTransaction", "eth_sign", "personal_sign"]
+            "events": ["accountsChanged", "chainChanged"]
+        },
+        "cosmos": {
+            ...
+        }
+    ],
```
Comment (FrederikBolding) [inline CAIPs/caip-25.md:61]:
```suggestion
    },
```

**Comment by bumblefudge** [comment]
open question, raised by @FrederikBolding : should the RPC call nest each of these provider objects named after namespaces into a parent array between `params` and `eip155/cosmos/etc`? i.e. 
```
{
  "id": 1,
  "jsonrpc": "2.0",
  "method": "provider_authorization",
  "params": {
    "namespaces": {
      "eip155": {
        "chains": [
          "eip155:1"
        ],
        "methods": [
          "eth_sendTransaction",
          "eth_signTransaction",
          "eth_sign",
          "personal_sign"
        ],
        "events": [
          "accountsChanged",
          "chainChanged"
        ]
      },
      "cosmos": {}
    }
  }
}
```

**Pull request merged**